engine/source/coupling/rad2rad/rad2rad_c.c

```diff
@@ -1123,7 +1188,45 @@ int *idp, *nng, *nodbuf;
 my_real_c *ms, *in;
 {send_mass_c(idp, nng, nodbuf, ms, in);}
 
+void send_mass_nl_c(idp, nng, iadd_nl, ms)
+int *idp, *nng, *iadd_nl;
+my_real_c *ms;
+{
+int i, j,nn, lbuf, flag;
+my_real_c *mbuf;
+
+    writer(fidw, (void *) idp, sizeof(int));    
+    readr(fidr, (void *) &flag, sizeof(int));
+    flagrot[*idp]=0;
+    
+    lbuf = *nng*sizeof(my_real_c);
+    mbuf = (my_real_c *) malloc(lbuf);
+
+    for (i = 0; i < *nng; i++)
+      {
+       mbuf[i] = ms[iadd_nl[i]-1];
+      }
+    
+    writer(fidw, (void *) mbuf, lbuf);    
+    free(mbuf);
```

Comment:
@bertrand-altair 

I think free of lbuf is missing

Reply:
lbuf is not an array